plugins/woocommerce-admin/client/marketplace/components/product-list-content/no-results.tsx

```diff
@@ -74,47 +82,68 @@ export default function NoResults(): JSX.Element {
 			return <></>;
 		}
 
+		const title = sprintf(
+			// translators: %s: product type (themes or extensions)
+			__( 'Most popular %s', 'woocommerce' ),
+			typeLabel
+		);
+
 		return (
 			<ProductList
-				title={ productGroup.title }
+				title={ title }
 				products={ productGroup.items }
 				groupURL={ productGroup.url }
 			/>
 		);
 	}
 
+	function getNoResultsIcon( type: ProductType ) {
+		if ( type === ProductType.theme ) {
+			return NoResultsThemesIcon;
+		}
+
+		return NoResultsExtensionsIcon;
+	}
+
 	return (
 		<div className="woocommerce-marketplace__no-results">
 			<div className="woocommerce-marketplace__no-results__content">
 				<img
 					className="woocommerce-marketplace__no-results__icon"
-					src={ NoResultsIcon }
+					src={ getNoResultsIcon( props.type ) }
 					alt={ __( 'No results.', 'woocommerce' ) }
 					width="80"
 					height="80"
 				/>
 				<div className="woocommerce-marketplace__no-results__description">
 					<h3 className="woocommerce-marketplace__no-results__description--bold">
 						{ sprintf(
-							// translators: %s: search term
+							// translators: %1$s product type (themes or extensions), %2$s: search term
 							__(
-								"We didn't find any results for “%s”",
+								"We didn't find %1$s for “%2$s”",
```

Comment:
I'm not sure how well this translates to other languages: "injecting" substrings like this can be tricky. I can't think of any specific examples, but there might exist languages where this doesn't work.

Like how in Japanese you use a different set of numbers when enumerating small quantities of physical things, depending on their shape, there could well be languages where the verb "find" conjugates differently depending on the object of the verb. Like if in English it was more-correct to say "We didn't find any themes" but "We didn't _discover_ any extensions".

Probably not an issue, and I can't think of a "nice" solution anyway, but just mentioning in case you think it's worth doing anything about.

Reply:
Should we just have two strings, one for themes and one for extensions?